### PR TITLE
[BREAKING] Change golang field representation to uint64

### PR DESCRIPTION
**Pull request by jeremyfelder**
## Describe the changes

This PR changes the field representation in Golang bindings to use uint64 limbs instead of uint32 for easier conversion with other libraries

### CODE REVIEW

File: wrappers/golang/internal/generator/main.go [hunk 1]
```diff
@@ -46,33 +46,33 @@ var bn254 = CurveData{
 	PackageName:    "bn254",
 	Curve:          "bn254",
 	GnarkImport:    "bn254",
-	ScalarLimbsNum: 8,
-	BaseLimbsNum:   8,
-	G2BaseLimbsNum: 16,
+	ScalarLimbsNum: 4,
+	BaseLimbsNum:   4,
+	G2BaseLimbsNum: 8,
```
Comment (DmytroTym) [inline wrappers/golang/internal/generator/main.go:51]: Just wondering: can this `CurveData` struct accommodate curves without G2 (like grumpkin)?

**Comment by DmytroTym** [comment]
Overall I'm not super good at golang or our golang bindings, but the PR looks pretty harmless to me and I can approve if needed.

File: wrappers/golang/internal/generator/main.go (hunk 1 above)
Comment (jeremyfelder) [reply wrappers/golang/internal/generator/main.go:51]: Currently no. The rest of the generation script for the curves also doesn't handle this case.
I think the entire generation script here needs to be a bit more modular in general and especially to support curves that don't implement all the functionality

Comment (DmytroTym) [reply wrappers/golang/internal/generator/main.go:51]: Right, cos grumpkin and some other curves also don't implement NTT (Nova doesn't need it) and we'll get to small fields hopefully soon where you don't have curves at all. But all that is outside of the scope of this PR ofc, just wanted to mention

File: wrappers/golang/core/internal/curve.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ func (p MockProjective) Size() int {
 	return p.X.Size() * 3
 }
 
-func (p MockProjective) AsPointer() *uint32 {
```
Comment (ImmanuelSegol) [inline wrappers/golang/core/internal/curve.go:11]: looks good to me, this was a long awaited change

**Review by ImmanuelSegol — APPROVED** [review]

**Review by ImmanuelSegol — APPROVED** [review]
Only question I have is how do we roll this out since its going to be a breaking change for many integrations.

File: wrappers/golang/core/internal/field.go [hunk 3]
```diff
@@ -6,30 +6,30 @@ import (
 )
 
 const (
-	BASE_LIMBS int8 = 8
+	BASE_LIMBS int = 4
```
Comment (bigsky77) [inline wrappers/golang/core/internal/field.go:9]: Limbs sum to 256 bytes.

8 * 32 = 256 
and 
4 * 64 = 256

Is this correct?

**Comment by bigsky77** [comment]
This shouldn't break any current integrations, all the changes make sense, and the tests pass.  Approve.

**Review by bigsky77 — APPROVED** [review]

File: wrappers/golang/core/internal/helpers_test.go [hunk 4]
```diff
@@ -4,27 +4,27 @@ import (
...
-	baseBytes := []byte{1, 2, 3, 4}
+func generateBytesArray(size int) ([]byte, []uint64) {
+	baseBytes := []byte{1, 2, 3, 4, 5, 6, 7, 8}
 	var bytes []byte
-	var limbs []uint32
+	var limbs []uint64
 	for i := 0; i < size; i++ {
 		bytes = append(bytes, baseBytes...)
-		limbs = append(limbs, 67305985)
+		limbs = append(limbs, 578437695752307201)
```
Comment (nonam3e) [inline wrappers/golang/core/internal/helpers_test.go:27]: what is this number for?

**Review by nonam3e — APPROVED** [review]

File: wrappers/golang/core/internal/helpers_test.go (hunk 4 above)
Comment (jeremyfelder) [reply wrappers/golang/core/internal/helpers_test.go:27]: Its the decimal representation of baseBytes interpreted as a uint64

**Pull request merged**